apps/hash-graph/libs/graph/src/store/postgres/ontology/data_type.rs

```diff
@@ -453,67 +490,60 @@ where
             .attach_printable("Could not read parent data types")?
             .data_types
             .into_iter()
-            .chain(
-                // ... and their parents (recursively)
-                transaction
-                    .get_data_types(actor_id, GetDataTypesParams {
-                        filter: Filter::for_data_type_parents(&required_parent_ids, None),
-                        temporal_axes: QueryTemporalAxesUnresolved::DecisionTime {
-                            pinned: PinnedTemporalAxisUnresolved::new(None),
-                            variable: VariableTemporalAxisUnresolved::new(None, None),
-                        },
-                        include_drafts: false,
-                        after: None,
-                        limit: None,
-                        include_count: false,
-                    })
-                    .await
-                    .change_context(InsertionError)
-                    .attach_printable("Could not read parent data types")?
-                    .data_types,
-            )
-            .for_each(|data_type| {
-                ontology_type_resolver.add_open(Arc::new(data_type.schema));
```

Comment:
We use the data from the database to calculate the inheritance chain for new types